shell/platform/fuchsia/dart-pkg/zircon_ffi/lib/zircon_ffi.dart

```diff
@@ -0,0 +1,36 @@
+// AUTO GENERATED FILE, DO NOT EDIT.
+//
+// Generated by `package:ffigen`.
+import 'dart:ffi' as ffi;
+
+/// Bindings for `dart:zircon_ffi`.
+class ZirconFFIBindings {
+  /// Holds the symbol lookup function.
+  final ffi.Pointer<T> Function<T extends ffi.NativeType>(String symbolName)
+      _lookup;
+
+  /// The symbols are looked up in [dynamicLibrary].
+  ZirconFFIBindings(ffi.DynamicLibrary dynamicLibrary)
+      : _lookup = dynamicLibrary.lookup;
+
+  /// The symbols are looked up with [lookup].
+  ZirconFFIBindings.fromLookup(
+      ffi.Pointer<T> Function<T extends ffi.NativeType>(String symbolName)
+          lookup)
+      : _lookup = lookup;
+
+  int zircon_dart_clock_get_monotonic() {
+    return _zircon_dart_clock_get_monotonic();
+  }
+
+  late final _zircon_dart_clock_get_monotonic_ptr =
+      _lookup<ffi.NativeFunction<_c_zircon_dart_clock_get_monotonic>>(
+          'zircon_dart_clock_get_monotonic');
+  late final _dart_zircon_dart_clock_get_monotonic
```

Comment:
@chinmaygarde, yup, I'm planning on building on top of this. Choosing a simple method like clock monotonic was the easiest way for me to test the combination of:

1. `ffigen` code generation within the engine repo.
2. Integration of `dart:zircon_ffi` with `dart:zircon`.
3. Testing this whole bundle in the fuchsia tree. (We do not test this code in the engine repo).

After this change, I wish to move `handle_waiter.dart` and `handle.dart` to the FFI model (away from tonic) as well.

Reply:
You run into symbol visibility issues and the linker or LTO thinking it can get rid of symbols with default visibility because they are not used. Kaushik and I were experimenting with it this morning and having a tough time making resolving symbols that way.

Reply:
I know this is already merged but I have experimented with this before and didn't need to do any extra trampolining by adding this method to the `package:zircon` instead of `dart:zircon` in fuchsia.git. The zx_* functions in libzircon get linked in at runtime by the os so you can get at this function  by just using `DynamicLibrary.process`.